libs/dotnet-sdk/UpdateStrategies/OnRequestUpdateStrategy.cs

```diff
@@ -0,0 +1,80 @@
+using Featureboard.DotnetSdk.Models;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System.Net;
+using System.Net.Http.Json;
+
+namespace Featureboard.DotnetSdk.UpdateStrategies;
+
+public class OnRequestUpdateStrategy : IFeatureBoardUpdateStrategy
+{
+  private readonly IFeatureboardState _state;
+  private readonly IOptions<FeatureboardOptions> _options;
+  private readonly IHttpClientFactory _httpClientFactory;
+  private readonly ILogger<OnRequestUpdateStrategy> _logger;
+
+  public OnRequestUpdateStrategy(IFeatureboardState state, IOptions<FeatureboardOptions> options, IHttpClientFactory httpClientFactory, ILogger<OnRequestUpdateStrategy> logger)
+  {
+    _state = state;
+    _options = options;
+    _httpClientFactory = httpClientFactory;
+    _logger = logger;
+  }
+
+  public async Task<FeatureConfiguration?> Get(string featureKey, CancellationToken cancellationToken)
+  {
+    await UpdateState(cancellationToken);
+    return await _state.Get(featureKey, cancellationToken);
+  }
+
+  private async Task UpdateState(CancellationToken cancellationToken)
```

Comment:
Maybe name this `UpdateWhenExpired` or something similar

Reply:
Code removed